src/storage/heap_file.c

```diff
@@ -10635,12 +10428,14 @@ heap_get_class_name_alloc_if_diff (THREAD_ENTRY * thread_p, const OID * class_oi
   char *copy_classname = NULL;
   RECDES recdes;
   HEAP_SCANCACHE scan_cache;
-  OID root_oid;
+  OID root_oid = OID_INITIALIZER;
+  HEAP_GET_CONTEXT context;
+
+  heap_init_get_context (&context, class_oid, &root_oid, &recdes);
 
   heap_scancache_quick_start_root_hfid (thread_p, &scan_cache);
 
-  if (heap_get_with_class_oid (thread_p, &root_oid, class_oid, &recdes, &scan_cache, S_SELECT, PEEK,
-			       LOG_ERROR_IF_DELETED) == S_SUCCESS)
+  if (heap_get_last_version (thread_p, &context, &scan_cache, PEEK, NULL_CHN) == S_SUCCESS)
```

Comment:
heap_get_last_version here must be different than the function used in locator_lock_and_get_object.

here it seems to me it is enough to call heap_get_visible (which can handle non-mvcc classes to including root class).